### ISSUE TITLE
monitor streams

### ISSUE THREAD

**Issue by StevenOrnico** [StevenOrnico/tvrecorder#1]
@wakoJackal  ffprobe on streams

**Comment by NyxJackal** [comment]
Okay. Been going through the files to see how they work. I'm on it. :)

**Comment by StevenOrnico** [comment]
thanks, I just wanted to make sure you had access to the project

**Comment by NyxJackal** [comment]
Hey, things are going well here. I wanted to ask, what do you have in mind for the front end other staff members will use? It will be built on the flask framework but what info would you like them to be able to access?

**Comment by StevenOrnico** [comment]
the system is a Disaster Recovery solution. We can scale up the system by adding additional PCI-e tv tuners. typically we can record about 8 channels per card, these are currently configured in the config files, these correspond to the channels found here:https://www.lyngsat.com/packages/DSTV-South-Africa.html

![image](https://user-images.githubusercontent.com/54528854/64171307-a2792800-ce52-11e9-848e-b0104151a10d.png)

corresponds with this 

![image](https://user-images.githubusercontent.com/54528854/64171340-b45acb00-ce52-11e9-82fe-cdd479d332ed.png)

the front end I only need it to be a faster way (web access) to managing/restarting/checking the recording status. to "glue" the system together

I hope this makes sense

**Comment by NyxJackal** [comment]
Kinda does. I'll send you something soon and if it isn't what you like we can build from there.

**Comment by StevenOrnico** [comment]
Thanks, I am trying to get Oresti no the platform so we don't have to doubt communicate on the finer details

**Comment by NyxJackal** [comment]
Okay. I will keep an eye out for it.

**Comment by NyxJackal** [comment]
By the way, any history using mencoder? It is one of the suggestions on clear recordings.

**Comment by StevenOrnico** [comment]
Hi, I do believe we have not had any experience with MEncoder

**Comment by NyxJackal** [comment]
Okay. Some have mentioned that it might be better. I will write some commands for it and maybe then you could test them out. How does that sound to you?

**Comment by Orestaki** [comment]
Hi guys, I am now on this platform as well. I have read through the comments and am uptodate.

**Comment by NyxJackal** [comment]
Welcome, Orestaki. I have uploaded some test files for both ffmpeg and mencoder with a link to download mencoder for yourself. Since I don't have access to a TV card, olease let me know if there are any bugs.

**Comment by Orestaki** [comment]
@StevenOrnico   Any news?

**Comment by StevenOrnico** [comment]
@wakoJackal I see there are 2 files, do I replace the current record_all.sh file with the one supplied and see the outcome? Is there any prerequisite for this to work.

Perhaps I can give you access to our dev machine to allow for faster dev time? Are you familiar with Linux Ubuntu 18.04 severs?

**Comment by NyxJackal** [comment]
Yes, they goal was to replace the record_all.sh file with the zipped files one at a time and see how each runs. Mencoder would need to be downloaded for the mencoder zipped file.

Yes, I have worked with Linux before. I think access to the server would make things quicker, if you don't mind. Maybe I could access it at a time that would be more favourable for you.

**Comment by StevenOrnico** [comment]
Apologies for the lack of communication. I had to take some personal time off work. I will grant you access to the machines via ssh tomorrow and provide details for it ASAP. I have pulled the updates to the project and am attempting to test the modification. I will provide feedback ASAP

**Comment by NyxJackal** [comment]
Okay, thank you.

**Comment by StevenOrnico** [comment]
Hi

the machine is available on 105.27.175.218:2222 via ssh

username:dev
password:Z3x3sk1ng413

after 3 failed login attempts you would be blocked for 20 min

The current project is located at /home/stevenp/tv

**Comment by StevenOrnico** [comment]
I have pulled the lastes code to that location

**Comment by NyxJackal** [comment]
Thank you. I will get started on it and give you a progress report tomorrow.

**Comment by NyxJackal** [comment]
Hey, I have tried to access the server via ssh on 2 different servers and failed. I got a time out response. When I tried to ping it, it was unsuccessful with a 100% packet loss.

**Comment by NyxJackal** [comment]
Though I could have missed something.

**Comment by StevenOrnico** [comment]
I will test from my external source ASAP and advice

**Comment by StevenOrnico** [comment]
Apologies our fiber line is currently down i have logged a ticket with sea-com and will let you know as soon as the issue has been resolved

**Comment by StevenOrnico** [comment]
Hi, @wakoJackal I have confirmed those details do work now from an external IP address. I was able to access the server using those details once the line was repaired.

![image](https://user-images.githubusercontent.com/54528854/65215873-616a5000-daaf-11e9-8f1c-dc5d60cb9260.png)

I use putty to ssh in and remember the password is case sensitive. Please let me know if you have any issues

**Comment by NyxJackal** [comment]
Thank you, I will do so.

**Comment by Orestaki** [comment]
Hi @wakoJackal   Has there been any progress?

**Comment by NyxJackal** [comment]
Yes. I have added mencoder to the machine and it is running as it should be. I have mainly been running tests on the machine to see what is causing the delay. I think I have some ideas of some of the causes. 

I will keep running tests and adjusting the code during the weekend.

**Comment by NyxJackal** [comment]
Good afternoon Steve. While I have been able to test ffmpeg and mencoder as to their functionality, the tests on the TV card have failed. Accessing the TV card remotely is the cause of this - I could be missing something, but so far remote access to the card is apparently not possible.

**Comment by StevenOrnico** [comment]
Please describe the process you initiate the tv card and then record? 

I believe you start the recording part of our software and then use DVblast to start and manage the TV Card which then streams to the local internal network

**Comment by NyxJackal** [comment]
Ah, okay. I had been writing the code in the command line and testing it there. Thank you - I have already began re running the tests by using the method you guys used.

**Comment by StevenOrnico** [comment]
Please let us know if you have had any success on this method.

**Comment by NyxJackal** [comment]
Good morning. No, it has not worked out. I used the method you suggested (and mentioned in the README.md) and replaced the code in the record_channel.sh file with the original one, then with variations on my own code and it didn't give any results. A few minutes ago I just tried the original code and changed the segment length from an hour to a minute and so far it has not worked out.

I may be wrong but I think it has something to do with the fact that I would be accessing the TV card via ssh.  While running ffmpeg and the like via ssh are doable, based on what others have to say on the internet, accessing a TV card using this method is not.

**Comment by Orestaki** [comment]
Hi @StevenOrnico Please can you advise as to next steps.
